src/anomaly/automl_regressors.rs

```diff
@@ -0,0 +1,161 @@
+//! Meta-ML regressors built on the [automl] crate
+
+use automl::{
+    settings::{Algorithm, DecisionTreeRegressorParameters},
+    SupervisedModel,
+};
+use smartcore::dataset::Dataset;
+
+use super::metaml::{MetaMLDataset, MetaMLModel};
+
+// Transforms a MetaMLDataset into an automl dataset
+fn metaml_dataset_to_automl(data: MetaMLDataset) -> Dataset<f32, f32> {
+    // Double-check meta-information about the data
+    assert!(data.features.ncols() == 6);
+    assert!(data.features.nrows() == data.targets.len());
+
+    // Data labels
+    let feature_names = vec![
+        "Cardinality".to_string(),
+        "Radius".to_string(),
+        "Local fractal dimension".to_string(),
+        "Cardinality (exponential moving average)".to_string(),
+        "Radius (exponential moving average)".to_string(),
+        "Local fractal dimension (exponential moving average)".to_string(),
+    ];
+    let target_names = vec!["Score".to_string()];
+    let description = "A dataset for training a meta-ml model to predict ROC AUC based on six inputs".to_string();
+
+    // Construct and return a dataset
+    Dataset {
+        num_samples: data.features.nrows(),
+        num_features: 6,
+        feature_names,
+        target_names,
+        description,
+        data: data.features.into_raw_vec(),
+        target: data.targets.into_raw_vec(),
+    }
+}
+
+#[derive(Default)]
+/// A metaml wrapper for an [automl] linear regressor
+pub struct LinearRegressor {
+    model: Option<SupervisedModel>,
+}
+
+impl LinearRegressor {
+    /// Constructs a new LinearRegressor
```

Comment:
Instead of documenting the `new` method, move this documentation to the struct itself.